### PR TITLE
some final contract changes

### PR DESCRIPTION
**Pull request by tehnalogos**

### CODE REVIEW

File: solidity/contracts/ArchiveHelpers.sol [hunk 1]
```diff
@@ -16,13 +16,19 @@ interface IFractal {
...
+    ) external returns (address) {
         FractalClone fractalClone = new FractalClone(registry, codehub, seed);
         return address(fractalClone);
     }
 
-    function fibonacciIterations(uint256 n) external pure returns (uint256) {
-        uint256 multiplier = 1000;
+    function fibonacciIterations(
+        uint256 n,
+        uint256 multiplier
```
Comment (tehnalogos) [inline solidity/contracts/ArchiveHelpers.sol:30]: only thing that changed in this file besides formatting

File: frontend/components/ContributionsRow.tsx [hunk 2]
```diff
@@ -117,14 +120,17 @@ export const ContributionsRow = ({
...
-          all = all.filter(
-            (v, i, a) => a.findIndex((t) => t.id === v.id) === i,
-          );
+          const all = [...(Array.isArray(prevArchives) ? prevArchives : [])];
+          for (const archive of newArchives) {
+            const oldIndex = all.findIndex((t) => t.id === archive.id);
+            if (oldIndex !== -1) {
+              all[oldIndex] = {
+                ...archive,
+              };
```
Comment (Ledespana) [inline frontend/components/ContributionsRow.tsx:129]: I'm confused about this. can you elaborate why?

File: frontend/components/wallet/WalletProvider.tsx [hunk 3]
```diff
@@ -191,15 +191,24 @@ export const WalletProvider: React.FC<Props> = ({ children }) => {
...
       contract.on("RefineToArchive", (sender, value) => {
         console.log("REFINE TO ARCHIVE EVENT", sender, value);
-        setRefineEventCounter((prevCounter) => {
+        setUserActionCounter((prevCounter) => {
+          return prevCounter + 1;
+        });
+      });
+      contract.on("Transfer", (sender, value) => {
+        console.log("TRANSFER EVENT", sender, value);
+        setUserActionCounter((prevCounter) => {
```
Comment (Ledespana) [inline frontend/components/wallet/WalletProvider.tsx:203]: this is going to duplicate per second the amount of calls we make. Do we really need this?

File: frontend/components/ContributionsRow.tsx (hunk 2 above)
Comment (tehnalogos) [reply frontend/components/ContributionsRow.tsx:129]: new archive object might have same id but updated properties for who the owner is and whether it's minted of not @Ledespana

File: frontend/components/wallet/WalletProvider.tsx (hunk 3 above)
Comment (tehnalogos) [reply frontend/components/wallet/WalletProvider.tsx:203]: once minting the interface does not update stats so it makes it confusing from user perspective @Ledespana

**Review by Ledespana — APPROVED** [review]

**Pull request merged**